contracts/Identity.sol

```diff
@@ -7,16 +7,48 @@ import "openzeppelin-eth/contracts/cryptography/ECDSA.sol";
 
 contract Identity is KeyManager {
   using ECDSA for bytes32;
-
-  constructor(address owner) public {
+  /**
+  * @dev Create Identity and set default keys
+  * @param managementKey address value for management key. This is the owner
+  * on behalf of this identity
+  * @param keys bytes32[] keys to be added to the identity
+  * @param purposes uint256[] purposes to be added to the identity
+  */
+  constructor(
+    address managementKey,
+    bytes32[] memory keys,
+    uint256[] memory purposes
+  )
+  public
+  {
     // Add MANAGEMENT_KEY
-    bytes32 key_ = addressToKey(owner);
-    _keys[key_].purposes.push(1);
-    _keys[key_].keyType = 1;
-    _keysByPurpose[1].push(key_);
-  }
+    bytes32 managementKey_ = addressToKey(managementKey);
+    _keys[managementKey_].purposes.push(MANAGEMENT);
+    _keys[managementKey_].keyType = 1;
+    _keysByPurpose[MANAGEMENT].push(managementKey_);
 
 
+    require(
+      keys.length == purposes.length,
+      "Keys and purposes must have the same length"
+    );
+
+    // Add general purpose keys
+    for (uint i = 0; i < purposes.length; i++) {
+      bytes32 keyToAdd_ = keys[i];
+
+      require(
+        purposes[i] != MANAGEMENT,
+        "Constructor can not add management keys"
+      );
+
+      _keys[keyToAdd_].purposes.push(purposes[i]);
```

Comment:
We are not checking if the same key appears twice in the `_keys` array.
I think that's fine because maybe I would like to add multiple purposes for a key.

What happens If I add the same purpose twice? Would that be a problem?

Reply:
The worst think it can happen it to have duplicated purposes in the array

Reply:
Exactly. Would that be a problem?

Reply:
Nope, because you can not remove a key purpose, you can just remove a key